### DISCUSSION TITLE
X11iraf - missing X11/Intrinsic.h

### DISCUSSION THREAD

**Discussion by WCKeel** [iraf-community/iraf#258] [Q&A] [answered]
Trying to build X11 tools (xgterm, of course) on MacOS 12.6 (M1 MacBoo Pro). After make, I get
/Library/Developer/CommandLineTools/usr/bin/make -C obm
cc -I/usr/include/tcl -I/usr/X11/include   -c -o Obm.o Obm.c
In file included from Obm.c:5:
./ObmP.h:16:10: fatal error: 'X11/Intrinsic.h' file not found
#include <X11/Intrinsic.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [Obm.o] Error 1
make: *** [obm/libobm.a] Error 2

Any hints?

**Comment by olebole** [comment, marked as answer]
Did you install the X11 package from https://www.xquartz.org/?

**Comment by WCKeel** [comment]
> On Feb 9, 2023, at 2:11 PM, Ole Streicher ***@***.***> wrote:
> 
> 
> Did you install the X11 package from https://www.xquartz.org/ <https://www.xquartz.org/>?
> 
> 

I thought I had, but in fact had not (setup of this machine has been broken up by a lot of other things over months and they say memory is the first thing to go.

I am now discovering the wonders of root in recent MacOS versions…

Thanks!

Bill Keel

--------------------------------------------------------------------------------------
William C. Keel                                        205-348-1641 (office)
Professor, Physics and Astronomy      205-348-5051 (fax)
Box 870324                                              205-348-5050 (dept.)
University of Alabama                             http://astronomy.ua.edu/keel <http://astronomy.ua.edu/keel>
Tuscaloosa, AL 35487-0324, U.S.A.    ***@***.*** ***@***.***>
--------------------------------------------------------------------------------------

**Comment by WCKeel** [comment]
Bit if a followup - how can I enable backspace/up arrow, etc. in this xgterm? Options in the X11 preferences do not look helpful. (I checked from both bash and tcsh).

**Reply by olebole** [reply]
It should work in xgterm. For reference, you can also start xterm and check if the behavior is different. However, to have it enabled in IRAF, you need to make sure to start **ecl**, not **cl**.

**Comment by WCKeel** [comment]
> On Feb 11, 2023, at 1:49 PM, Ole Streicher ***@***.***> wrote:
> 
> 
> It should work in xgterm. For reference, you can also start xterm and check if the behavior is different. However, to have it enabled in IRAF, you need to make sure to start ecl, not cl.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/orgs/iraf-community/discussions/258#discussioncomment-4945278>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A5ZAVNR67FM7M2WMT2XGH7LWW7UMXANCNFSM6AAAAAAUW2EWTM>.
> You are receiving this because you authored the thread.


That was what I was missing - it took me additional searching to find out to use ecl. So now the arrows and erase work.

Thanks,

Bill Keel
--------------------------------------------------------------------------------------
William C. Keel                                        205-348-1641 (office)
Professor, Physics and Astronomy      205-348-5051 (fax)
Box 870324                                              205-348-5050 (dept.)
University of Alabama                             http://astronomy.ua.edu/keel <http://astronomy.ua.edu/keel>
Tuscaloosa, AL 35487-0324, U.S.A.    ***@***.*** ***@***.***>
--------------------------------------------------------------------------------------
